src/main/java/com/americanexpress/span/core/SPANInitialization.java

```diff
@@ -31,50 +32,50 @@
 
 public final class SPANInitialization {
 
-    private SPANInitialization(){}
+    private SPANInitialization() {
+    }
 
-    private static final String APP_PROFILE = "APP_PROFILE";
-    private static final String DEFAULT_FILE_NAME = "SPANConfig";
-    private static final String YAML = ".yaml";
+    private static PropertyConfiguration propertyConfiguration = new PropertyConfiguration() {};
 
     /**
      * Loads the given yaml file into SPANConfig object. If no fileName given, loads the default file.
      *
-     * @param fileName
+     * @param configuration
      * @throws ConfigurationSPANException
      */
-    public static final void initialize(String fileName) {
-
-        if (StringUtils.isEmpty(fileName)) {
+    public static final void initialize(PropertyConfiguration configuration) {
+        if (Objects.nonNull(configuration)) {
+            propertyConfiguration = configuration;
+        }
+        if (StringUtils.isEmpty(propertyConfiguration.getFileName()) || propertyConfiguration.getFileName().equals(SPANConstants.DEFAULT_FILE_NAME)) {
```

Comment:
This statement is to check 2 negative scenarios:
1. When application/user override getFileName()  method and return an empty/blank/null
2. When application/user override only getAppProfileEnvVariable(). So on default  getFileName() return "SPANConfig.yaml". So if we don't have this check -propertyConfiguration.getFileName().equals(SPANConstants.DEFAULT_FILE_NAME), then it wouldn't call initialize() method, and use the environment variable.

Reply:
Please rearrange the condition checking , it should be SPANConstants.DEFAULT_FILE_NAME.equals(...)

Reply:
@shailendrasoni02 This code change is outdated. I have made changes on this logic on Friday as per our discussion on Thursday. Now the default method returns null unless user override it (PropertyConfiguration line 13)-
   default String getSPANConfigFileName() {
        return null;
    }

So we are no longer checking if propertyConfiguration.getFileName().equals(SPANConstants.DEFAULT_FILE_NAME).
Below is the current code in SPANInitialization
if (StringUtils.isEmpty(propertyConfiguration.getSPANConfigFileName())) {
            initialize();
        } else {
            //Load the configuration file.
            loadSPANConfig(propertyConfiguration.getSPANConfigFileName());
        }